### ISSUE TITLE
group-to-group relationships are not multi-to-multi as they should be

### ISSUE THREAD

**Issue by RusDavies** [RusDavies/moose-inventory#6]
The group-of-groups model is borked.  A parent group should be able to have multiple children, and a child group multiple parents.   However, in the current model, each child may have only 1 parent.  Example follows.

Add a parent and two child groups, and associate them:  

```
moose-inventory group add parent1 child1 child2 
moose-inventory group addchild parent1 child1 child2 
```

Now list the groups: 

```
$ moose-inventory group list
{
  "parent1": {
    "children": [
      "child1",
      "child2"
    ]
  },
  "child1": {
  },
  "child2": {
  }
}
```

Now add another parent, and associate it with the same two children

```
moose-inventory group add parent2
moose-inventory group addchild parent2 child1 child2 
```

If we list the groups again, then, rather than both children being associated with both parents, instead the creation of associations with parent2 silently destroyed those with parent1. 

```
$ moose-inventory group list
 {
  "parent1": {
  },
  "parent2": {
    "children": [
      "child1",
      "child2"
    ]
  },
  "child1": {
  },
  "child2": {
  }
}
```

**Comment by RusDavies** [comment]
Problem comes from the definition of the model for the groups table:

``` ruby
# Model for the groups table
class Group < Sequel::Model
  many_to_one :parent, :class=>self
  one_to_many :children, :key=>:parent_id, :class=>self
  many_to_many :hosts
  one_to_many :groupvars
end
```

and:

``` ruby
unless @db.table_exists? :groups
  @db.create_table(:groups) do
    primary_key :id
    foreign_key :parent_id
    column :name, :text, unique: true
  end
end
```

Clearly, this is set up as a one-to-many relationship between parent and children.

**Comment by RusDavies** [comment]
Fixed in 1.0.1 

New code is as follows: 

``` ruby
# Model for the groups table
class Group < Sequel::Model
  many_to_many :parents,  
    :left_key=>:parent_id, 
     :right_key=>:child_id, 
     :class=>self

  many_to_many :children, 
    :left_key=>:child_id, 
    :right_key=>:parent_id, 
    :class=>self        

    many_to_many :hosts 
    one_to_many  :groupvars
end   
```

and 

``` ruby
unless @db.table_exists? :groups
  @db.create_table(:groups) do
    primary_key :id
    column :name, :text, unique: true
  end
end

unless @db.table_exists? :groups_groups
  @db.create_table(:groups_groups) do
    primary_key :id
    foreign_key :parent_id, :groups
    foreign_key :child_id,  :groups
  end
end
```
